Tools/Scripts/webkitpy/xcode/simulated_device.py

```diff
@@ -420,30 +436,44 @@ def initialize_devices(cls, requests, host=None, name_base='Managed', simulator_
         if not hasattr(requests, '__iter__'):
             requests = [requests]
 
+        # Parse user-specified UDIDs
+        if type(udids) is str:
+            udids = udids.split(',')
+
         # Check running sims
+        deferred_booted_devices = []
         for device in cls.available_devices(host):
-            matched_request = cls._does_fulfill_request(device, requests)
+            matched_request = cls._does_fulfill_request(device, requests, True)
             if matched_request is None:
                 continue
-            requests.remove(matched_request)
-            _log.debug(u'Attached to running simulator {}'.format(device))
-            SimulatedDeviceManager.INITIALIZED_DEVICES.append(device)
 
-            # DeviceRequests are compared by reference
-            requests_copy = [request for request in requests]
+            device_is_booted = device.platform_device.is_booted_or_booting()
+            if udids is not None and device.platform_device.udid not in udids:
```

Comment:
nit: `if udids` should suffice - same for line 465

Reply:
Thanks for the tip! Resolved in latest commit.